### PR TITLE
Use NuGetTargetMoniker instead of TFM if its exists

### PR DESCRIPTION
**Pull request by davkean**
Fixes: https://github.com/dotnet/project-system/issues/4854.

Note: This has localization impact that I believe is okay; if we do not ship localized values - then the user will continue to get the old string.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilderTests.cs [hunk 1]
```diff
@@ -62,6 +62,27 @@ public void ToProjectRestoreInfo_WhenNuGetRestoreRuleMissing_ReturnsEmpty()
             Assert.Empty(targetFramework.Properties);
         }
 
+        [Fact]
+        public void ToProjectRestoreInfo_RespectsNuGetTargetMonikerIfPresent()
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilderTests.cs]: Minor: Cross ref GitHub issue here for future reference: https://github.com/dotnet/project-system/issues/4854

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilderTests.cs]: You don't think history gives you that?

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilderTests.cs]: Sure it does. Just not as convenient.

**Comment by tmeschter** [comment]
Non-english users will continue to see the older (but still localized) messages, yes.

**Comment by jamesmontemagno** [comment]
Is this in 16.1.1 or future?

**Comment by davkean** [comment]
16.1.2

**Pull request merged**